### PR TITLE
Return ResourceWrapper without pointer

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Worker in async cache panics because it fails to cast resource to `*ResourceWrapper`

cache (*Resource) -> cast to (*Resource) // it works

s3 (Resource) -> cast to (*Resource) // Failed to do type casting since we unmarshall the resource to `Resource` [here](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/internal/webapi/core.go#L50). 

We read the resource from s3 bucket after restarting the propeller (cache becomes empty). 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to https://github.com/flyteorg/flyte/pull/4125. Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to https://github.com/flyteorg/flyte/pull/4132. Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.
